File: app/controllers/HomeController.java
package controllers;

import akka.japi.Pair;
import models.ScrapedDataModel;
import play.mvc.*;
import scala.Tuple2;
import scala.collection.immutable.List;
import utils.Downloader;
import utils.readers.JsonConfigReader;

import java.util.ArrayList;

public class HomeController extends Controller {
    public Result scraper() {
        var configReader = new JsonConfigReader("nike_scrapper.json");
        configReader.getConfig();
        List<ScrapedDataModel> scrappedData = Downloader.getHtmlDocumentsList(configReader.MultipleConfigurations());
        ArrayList<Tuple2<String, String>> scrapedString = new ArrayList<>();
        for (int i = 0; i < scrappedData.length(); i++ ) {
            var temp = scrappedData.apply(i).getPrettyStringMap();
            for (int j = 0; j < temp.length(); j++ ) {
                scrapedString.add(temp.apply(j));
            }
        }
        return ok(views.html.scraperResults.render(scrapedString));
    }
}

File: app/controllers/HomeController.scala
package controllers

import java.util

import javax.inject.{Inject, Singleton}
import models.ScrapedDataModel
import play.api.mvc._
import utils.Downloader
import utils.readers.JsonConfigReader

import scala.collection.immutable.List
import scala.collection.mutable.ListBuffer;

/**
 * This controller creates an `Action` to handle HTTP requests to the
 * application's home page.
 */
@Singleton
class HomeController @Inject()(cc: ControllerComponents) extends AbstractController(cc) {

  def scraper(): Action[AnyContent] = Action { implicit request: Request[AnyContent] =>
    val configReader: JsonConfigReader = new JsonConfigReader("nike_scrapper.json")
    configReader.getConfig()
    val scrappedData: List[ScrapedDataModel] = Downloader.getHtmlDocumentsList(configReader.MultipleConfigurations)
    val scrapedString: ListBuffer[(String, String)] = ListBuffer[(String, String)]()
    for (i <- scrappedData.indices) {
      val temp: List[(String, String)] = scrappedData.apply(i).getPrettyStringMap
      for (j <- temp.indices) {
        scrapedString.append(temp.apply(j))
      }
    }
    Ok(views.html.scraperResults.render(scrapedString.toList))
  }
}